src/lib/specs-crawler.js

```diff
@@ -78,13 +79,16 @@ async function crawlSpec(spec, crawlOptions) {
         path.dirname(crawlOptions.fallback) : '';
 
     if (spec.error) {
-        return specOrFallback(spec, fallbackFolder, crawlOptions.fallbackData);
+        return specOrFallback(spec, fallbackFolder, crawlOptions.fallbackData?.results);
     }
 
     try {
-        const fallback = crawlOptions.fallbackData?.find(s => s.url === spec.url);
-        const etag = fallback?.crawlCacheInfo?.etag;
-        const lastModified = fallback?.crawlCacheInfo?.lastModified;
+        let fallback;
+        let cacheInfo = {};
+        if (crawlOptions.fallbackData?.crawler === `reffy-${reffyVersion}`) {
+          fallback = crawlOptions.fallbackData?.results?.find(s => s.url === spec.url);
+          cacheInfo = Object.assign({}, fallback?.crawlCacheInfo);
```

Comment:
I would still use the fallback data when versions don't match, so as not to end up with dangling extract files in case of processing errors. I would just not set `cacheInfo` to force the processing.

Reply:
hmm... but if the format of the extract files have changed for a reason or another, I'm not sure re-using them is a good idea

Reply:
But that situation is meant to be temporary. If we don't reuse them, the dangling extract files will be removed from the package, triggering a pre-release PR to drop them (and a minor bump).

Reply:
I'm not sure I understand; say we change the JSON schema in idlparsed extracts, the said extracts with the previous schema are no longer consumable, so we shouldn't pretend we have the relevant data for the said spec.

Reply:
Well, we've always been pretending that we have the relevant data. We keep dangling extract files in the repo for the time being in the hope that they will still be accurate next time processing succeeds. They appear in the curated branch (in your example, not reusing the extract files won't change anything, the potentially outdated idlparsed extracts won't be dropped from the repo and will continue to appear in the curated branch in particular) and they are released in the npm packages as well.

The fallback mechanism gives us a way to keep the links from `index.json` to these dangling extract files, so that we effectively no longer have any dangling extract files in the repo. That's great, I don't like dangling extract files because I don't know what they mean.

The release mechanism now only considers extract files that are linked from `index.json` (not strictly needed today, but will be needed as soon as we need to filter non-browser specs).

When the version of Reffy changes, if we don't reuse the extract files, the pre-release PR will say "these files have been removed", which is not fantastic because the right message should rather be "we have not up-to-date extract files for these specs, and don't know whether they are still accurate". Also, we won't be able to release a new version of a package until we're back to a crawl without error situation.

If we reuse the extract files, the pre-release PR will not report anything about these files, which is not fantastic either because then we could miss the fact that some of the extract files may no longer have the right structure.

Actually, I think that we need to update the pre-release PR job to include in the diff the list of specs for which we only have fallback data (those that have an `error` property). Even if the structure of the extract files has not changed, there may be good reasons not to release a package if we realize that an extract is too outdated. I'm just worried that we'll often end up with a list of >100 CSS specs. I'll look into it.

In any case, I would prefer to stick to a "no dangling extract files" principle.

Reply:
ok, now the fallback is only skipped for not-modified situations when crawler versions don't match